tsdownsample/downsampling_interface.py

```diff
@@ -1,54 +1,102 @@
+"""AbstractDownsampler interface-class, subclassed by concrete downsamplers."""
+
+__author__ = "Jeroen Van Der Donckt"
+
+import re
 import warnings
 from abc import ABC, abstractmethod
 from types import ModuleType
-from typing import Callable
+from typing import Callable, List, Optional, Tuple, Union
 
 import numpy as np
-import pandas as pd
 
 
-class DownsampleInterface(ABC):
-    def __init__(self, name: str) -> None:
+class AbstractDownsampler(ABC):
+    """AbstractDownsampler interface-class, subclassed by concrete downsamplers."""
+
+    def __init__(self, name: str, dtype_regex_list: Optional[List[str]] = None):
         self.name = name
+        self.dtype_regex_list = dtype_regex_list
```

Comment:
Shouldn't we split the `dtype` regex list into `x_dtype_regex_list` and `y_dtype_regex_list`

Reply:
@jvdd, this can be resolved, as you've splitted up the `dtype`